Products/CMFPlone/CatalogTool.py

```diff
@@ -398,19 +398,29 @@ def searchResults(self, query=None, **kw):
         # It also accepts a keyword argument show_inactive to disable
         # effectiveRange checking entirely even for those without portal
         # wide AccessInactivePortalContent permission.
+        #
+        # By default, this shows inactive content to users with the required
+        # permission, while hiding it from users without AccessInactivePortalContent.
+        # - To show inactive content to users without the portal wide
+        #   AccessInactivePortalContent permission, pass show_inactive=True
+        #   as either a keyword or query parameter.
+        # - To suppress inactive content for admins, pass show_inactive=False
+        #   as either a keyword or query parameter
 
         # Make sure any pending index tasks have been processed
         processQueue()
 
         kw = kw.copy()
-        show_inactive = kw.get("show_inactive", False)
-        if isinstance(query, dict) and not show_inactive:
-            show_inactive = "show_inactive" in query
+        show_inactive = kw.get("show_inactive", None)
```

Comment:
I know I could've just used `kw.get("show_inactive")` here and below, but since this logic depends so crucially on `None` being different from `False`, I preferred the explicit variant.